velox/functions/sparksql/fuzzer/CMakeLists.txt

```diff
@@ -12,15 +12,113 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Set up Spark connect
+file(
+  GLOB PROTO_FILES
+  RELATIVE ${PROJECT_SOURCE_DIR}
+  ${CMAKE_CURRENT_SOURCE_DIR}/proto/spark/connect/*.proto)
+foreach(PROTO ${PROTO_FILES})
+  get_filename_component(PROTO_DIR ${PROTO} DIRECTORY)
+  get_filename_component(PROTO_NAME ${PROTO} NAME_WE)
+  set(PROTO_OUTPUT_DIR ${PROJECT_BINARY_DIR}/${PROTO_DIR})
+  list(APPEND PROTO_SRCS "${PROTO_OUTPUT_DIR}/${PROTO_NAME}.pb.cc")
+  list(APPEND PROTO_HDRS "${PROTO_OUTPUT_DIR}/${PROTO_NAME}.pb.h")
+  list(APPEND GRPC_SRCS "${PROTO_OUTPUT_DIR}/${PROTO_NAME}.grpc.pb.cc")
+  list(APPEND GRPC_HDRS "${PROTO_OUTPUT_DIR}/${PROTO_NAME}.grpc.pb.h")
+  list(APPEND PROTO_FILES_FULL
+       "${PROJECT_SOURCE_DIR}/${PROTO_DIR}/${PROTO_NAME}.proto")
+endforeach()
+
+set(PROTO_OUTPUT_FILES ${PROTO_HDRS} ${PROTO_SRCS})
+set_source_files_properties(${PROTO_OUTPUT_FILES} PROPERTIES GENERATED TRUE)
+
+set(GRPC_OUTPUT_FILES ${GRPC_HDRS} ${GRPC_SRCS})
+set_source_files_properties(${GRPC_OUTPUT_FILES} PROPERTIES GENERATED TRUE)
+
+# Ensure that the option --proto_path is not given an empty argument
+foreach(PROTO_PATH ${PROJECT_SOURCE_DIR} ${Protobuf_INCLUDE_DIRS})
+  list(APPEND PROTO_PATH_ARGS --proto_path=${PROTO_PATH})
+endforeach()
+
+# Generate Spark connect hearders and sources
+add_custom_command(
+  OUTPUT ${PROTO_OUTPUT_FILES}
+  COMMAND protobuf::protoc ${PROTO_PATH_ARGS} --cpp_out ${CMAKE_BINARY_DIR}
+          ${PROTO_FILES_FULL}
+  DEPENDS protobuf::protoc
+  COMMENT "Running PROTO compiler"
+  VERBATIM)
+add_custom_target(spark_connect_proto ALL DEPENDS ${PROTO_OUTPUT_FILES})
+
+# Generate grpc headers and sources
+add_custom_command(
+  OUTPUT ${GRPC_OUTPUT_FILES}
+  COMMAND
+    protobuf::protoc ${PROTO_PATH_ARGS} --grpc_out=${CMAKE_BINARY_DIR}
+    --plugin=protoc-gen-grpc=$<TARGET_FILE:gRPC::grpc_cpp_plugin>
+    ${PROTO_FILES_FULL}
+  DEPENDS protobuf::protoc
+  COMMENT "Running gRPC C++ protocol buffer compiler"
+  VERBATIM)
+add_custom_target(spark_connect_grpc_proto ALL DEPENDS ${GRPC_OUTPUT_FILES})
```

Comment:
This is not a core feature of velox so `ALL`  should be removed and these targets should be added to the link libraries of the query runner, that way they will be run if that target will be run.  (I know the sub dir will only be added when testing is enabled but this still communicates the intent better.)

Reply:
Removed `ALL`. Below error is met when trying to link `spark_connect_grpc_proto` to the target, so I just removed it. Does it makes sense? Thanks.

> Target "spark_connect_proto" of type UTILITY may not be linked into another target.